provider/src/main/kotlin/au/com/dius/pact/provider/ProviderVerifier.kt

```diff
@@ -349,6 +361,57 @@ open class ProviderVerifier @JvmOverloads constructor (
     }
   }
 
+  @Suppress("TooGenericExceptionCaught")
+  override fun verifyResponseByFactory(
+    providerInfo: IProviderInfo,
+    consumer: IConsumerInfo,
+    interaction: Interaction,
+    interactionMessage: String,
+    failures: MutableMap<String, Any>,
+    pending: Boolean
+  ): VerificationResult {
+    val interactionId = interaction.interactionId.orEmpty()
+    try {
+      val factory = responseFactory!!
+      return if (interaction.isAsynchronousMessage()) {
+        verifyMessage(
+          factory,
+          interaction as MessageInteraction,
+          interactionId,
+          interactionMessage,
+          failures,
+          pending
+        )
+      } else {
+        val expectedResponse = (interaction as SynchronousRequestResponse).response
+        val response = factory.apply(interaction.description) as Map<String, Any>
+        val actualResponse = ProviderResponse(
+          response["statusCode"] as Int,
+          response["headers"] as Map<String, List<String>>,
+          ContentType.UNKNOWN,
```

Comment:
Could the factory function not also supply the content type?

Reply:
@uglyog i copied this logic from here https://github.com/pact-foundation/pact-jvm/blob/master/provider/src/main/kotlin/au/com/dius/pact/provider/ProviderVerifier.kt#L327

Do we want to update that in both places?

Reply:
No, don't worry about that, I will fix the other place.